cohttp-eio/src/client.ml

```diff
@@ -0,0 +1,112 @@
+module Buf_read = Eio.Buf_read
+module Buf_write = Eio.Buf_write
+
+type response = Http.Response.t * Buf_read.t
+type host = string * int option
+type resource_path = string
+type 'a conn = unit -> (host * Eio.Flow.two_way as 'a)
+
+type 'a body_disallowed_call =
+  ?version:Http.Version.t ->
+  ?headers:Http.Header.t ->
+  'a conn ->
+  resource_path ->
+  response
+(** [body_disallowed_call] denotes HTTP client calls where a request is not
+    allowed to have a request body. *)
+
+type 'a body_allowed_call =
+  ?version:Http.Version.t ->
+  ?headers:Http.Header.t ->
+  ?body:Body.t ->
+  'a conn ->
+  resource_path ->
+  response
+
+(* Request line https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.1 *)
+let write_request writer (meth, version, headers, resource_path, body) =
+  Buf_write.string writer (Http.Method.to_string meth);
+  Buf_write.char writer ' ';
+  Buf_write.string writer resource_path;
+  Buf_write.char writer ' ';
+  Buf_write.string writer (Http.Version.to_string version);
+  Buf_write.string writer "\r\n";
+  Rwer.write_headers writer headers;
+  Buf_write.string writer "\r\n";
+  Body.write_body writer body
+
+(* response parser *)
+
+let is_digit = function '0' .. '9' -> true | _ -> false
+
+let status_code =
+  let open Rwer in
+  let open Buf_read.Syntax in
+  let+ status = take_while1 is_digit in
+  Http.Status.of_int (int_of_string status)
+
+let reason_phrase =
+  Buf_read.take_while (function
+    | '\x21' .. '\x7E' | '\t' | ' ' -> true
+    | _ -> false)
+
+(* https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.2 *)
+let response buf_read =
+  let open Buf_read.Syntax in
+  match Buf_read.at_end_of_input buf_read with
+  | true -> Stdlib.raise_notrace End_of_file
+  | false ->
```

Comment:
Is this needed? It will raise `End_of_file` anyway when you try to read the version.

Reply:
Yes, This is removed now.